docs/proposals/api-grouping.md

```diff
@@ -0,0 +1,375 @@
+	policy // not currently available through API, but might be some day.
+
+/apis/abac/v1beta1
+	policy // not currently available through API, but might be some day.
+
+/apis/authorization/v1beta1
+	subjectAccessReview
+	selfSubjectAccessReview
+	localSubjectAccessReview
+
+/apis/componentconfig/v1alpha1/
+	kubeProxyConfiguration
+
+/apis/metrics/v1alpha1
+	rawNode
+	rawPod
+```
+
+## Upcoming Additions
+
+New top level objects being considered:
+- `template` and `parameters` are added by #18215.  Other things do not
+   depend directly on them.  They work as an independent template expansion
+   service.
+- `Node`, `Pod` metrics types are being added, tenatively to `metrics` group. (see #18824)
+- `PetSet`: https://github.com/kubernetes/kubernetes/pull/18016 
+- `heartbeat` by #14735
+- `externalServiceProvider` by #13216
+- `podSecurityPolicy` and maybe `PodSecurityPolicyBinding` by #7893 
+- `workflow` by  #18827
+- `scheduledJob` by #11980
+- `metadataPolicy` by #18262
+- `ingress` may change, and probably will move out of the main api group.
+- `podStatusResult` should maybe move to a kubelet API group?
+
+## Inter-Kind references
+
+I list some dependencies here.
+
+- `ScheduledJob` depends on `Workflow` and `Job`; `Workflow` depends on `Job`
+- `Job`, `ReplicationController`, `ReplicaSet`, `Deployment` and `DaemonSet` make `Pods` via PodTemplates.
+  - this is a strong dependency.
+- Pods that use PersistentVolumeClaimVolumeSource depend on PersistentVolumeClaim object (`ClaimName`).
+- PersistentVolumeClaims depend on PersistentVolume names.
+- PersistentVolumes depend on PersistentVolumeClaims via ClaimRef.
+- `GlusterfsVolumeSource` of a Pod depends on `Endpoints`.
+- Pods depend on `Secrets` several ways: `SecretVolumeSource`, EnvVars.
+- `Pods` depend on `ServiceAccounts` via `ServiceAccountName`.
+  - this seems like a pretty strong dependency.
+- `Pods` depend on `Nodes` in that `Pods` have a `NodeName`.
+- `Pods` with an `RBDVolumeSource` or `CephFSVolumeSource` or `FlexVolumeSource` depend on `Secrets` to store a secret.
+- `Pods` depend on `Secrets` for `imagePullSecrets`.
+- `ServiceAccounts` refer to `Secrets`.
+- `Pods` can depend on `ConfigMap` or `Secret` via env vars, etc.
+- `ReplicationControllers` depend on `Pods` via a `PodTemplate`.
+
+## Reasons to have more than 1 group
+
+Why do we not put everything into 1 group.   That would be easier.
+Well, clearly we need namespacing, and we cannot control the naming of
+addons,
+
+But, why not one group for "the project" and then add another group
+for any "third-party" APIs?
+
+Some reasons are:
+
+- Groups will eventually allow extension by third parties that we as a core project
+  would not accept.  We as a core project have to use the extension mechanism to keep
+  ourselves honest, though.
+
+- The same reasons programs have packages: to enforce boundaries, and prevent coupling (taken from #635).
+- To allow api versioning at different rates (taken from #635)
```

Comment:
Okay.  I called that out.  I need to think about what that implies for selecting the best grouping.